juriscraper/opinions/united_states/state/pa.py

```diff
@@ -5,74 +5,135 @@
 """
 
 import re
+from datetime import date, datetime, timedelta
+from typing import Dict, Tuple
+from urllib.parse import urlencode
 
-from juriscraper.lib.html_utils import get_xml_parsed_text
-from juriscraper.lib.string_utils import convert_date_string
+from juriscraper.AbstractSite import logger
+from juriscraper.lib.date_utils import make_date_range_tuples
 from juriscraper.OpinionSiteLinear import OpinionSiteLinear
 
 
 class Site(OpinionSiteLinear):
+    court = "Supreme"
+    base_url = "https://www.pacourts.us/api/opinion?"
+    document_url = "https://www.pacourts.us/assets/opinions/{}/out/{}"
+    days_interval = 50
+    api_dt_format = "%Y-%m-%dT00:00:00-05:00"
+    first_opinion_date = datetime(1998, 4, 27)
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.court_id = self.__module__
-        self.regex = False
-        self.url = "https://www.pacourts.us/Rss/Opinions/Supreme/"
-        self.set_regex(r"(.*)(?:[,-]?\s+Nos?\.)(.*)")
-        self.base = (
-            "//item[not(contains(title/text(), 'Judgment List'))]"
-            "[not(contains(title/text(), 'Reargument Table'))]"
-            "[not(contains(title/text(), 'Order Amending Rules'))]"
-            "[contains(title/text(), 'No.')]"
+        self.regex = re.compile(r"(.*)(?:[,-]?\s+Nos?\.)(.*)")
+        self.status = "Published"
+
+        now = datetime.now() + timedelta(days=1)
+        start = now - timedelta(days=7)
+        self.params = {
+            "startDate": start.strftime(self.api_dt_format),
+            "endDate": now.strftime(self.api_dt_format),
+            "courtType": self.court.upper(),
+            "boardDocketNumber": "undefined",
+            "courtDocketNumber": "undefined",
+            "keywords": "undefined",
+            "postTypes": "cd,co,do,mo,oaj",
```

Comment:
we should expand the post Types to 
`cd,cds,co,cs,dedc,do,ds,mo,oaj,pco,rv,sd`

PA does a weird thing where they split apart lots of things